zerver/tests/test_message_send.py

```diff
@@ -2609,6 +2609,57 @@ def test_ensure_stream_gets_called(self) -> None:
         # wasn't automatically created.
         Stream.objects.get(name=stream_name, realm_id=realm.id)
 
+    def test_direct_message_to_self_and_bot_in_dm_disabled_org(self) -> None:
+        """
+        Test that a user can send a direct message to themselves and to a bot in a DM disabled organization
+        """
+        user = self.example_user("hamlet")
+        user.realm.private_message_policy = Realm.PRIVATE_MESSAGE_POLICY_DISABLED
+        user.realm.save()
+
+        # Create a new bot user
+        bot = do_create_user(
+            email="test-bot@zulip.com",
+            password="",
+            realm=user.realm,
+            full_name="Test Bot",
+            bot_type=UserProfile.DEFAULT_BOT,
+            bot_owner=user,
+            acting_user=None,
+        )
+
+        # Test sending a message to self
+        result = self.api_post(
+            user,
+            "/api/v1/messages",
+            {
+                "type": "private",
+                "to": orjson.dumps([user.id]).decode(),
+                "content": "Test message to self",
+            },
+        )
+        self.assert_json_success(result)
+
+        msg = self.get_last_message()
+        expected = "Test message to self"
+        self.assertEqual(msg.content, expected)
+
+        # Test sending a message to the bot
+        result = self.api_post(
+            user,
+            "/api/v1/messages",
+            {
+                "type": "private",
+                "to": orjson.dumps([bot.id]).decode(),
+                "content": "Test message to bot",
+            },
+        )
+        self.assert_json_success(result)
+
+        msg = self.get_last_message()
+        expected = "Test message to bot"
+        self.assertEqual(msg.content, expected)
```

Comment:
This test should verify that trying to send the same message to another non-bot user fails.

Reply:
Now the test also verifies dm to not-bot-user users.